File: src/ios/CardLink.m
#import "CardLink.h"

@implementation CardLink

- (void)establishWSS:(CDVInvokedUrlCommand*)command{
    CDVPluginResult* pluginResult = nil;
    NSString* wssURL = [command.arguments objectAtIndex:0];

    if (wssURL != nil && [wssURL length] > 0) {
        [self.webSocketClientManager connectTo:url];
    
        if (self.webSocketClientManager.isConnected) {
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"true"];
        } else {
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
        }
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
    }

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end

File: ios/CardLink.swift
@objc(CardLink) class CardLink: CDVPlugin {

    @objc(establishWSS:)
    func establishWSS(command: CDVInvokedUrlCommand) {
        var pluginResult: CDVPluginResult? = nil
        
        // Überprüfen, ob der erste Parameter eine gültige URL ist
        if let wssURL = command.arguments.first as? String, !wssURL.isEmpty {
            // Verbindung mit der URL herstellen
            //self.webSocketClientManager.connectToURL(wssURL)
            
            // Überprüfen, ob die Verbindung erfolgreich hergestellt wurde
            //if self.webSocketClientManager.isConnected {
                //pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: "true")
            //} else {
                //pluginResult = CDVPluginResult(status: CDVCommandStatus_ERROR)
            //}

            pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: wssURL)
        } else {
            pluginResult = CDVPluginResult(status: CDVCommandStatus_ERROR)
        }

        // Das Ergebnis an den Cordova-Callback zurückgeben
        self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
    }
    
    // WebSocketClientManager-Instanz bereitstellen
    private var webSocketClientManager = WebSocketClientManager()
}
